src/index.js

```diff
@@ -0,0 +1,225 @@
+'use strict'
+
+const { Record } = require('libp2p-record')
+const { Key } = require('interface-datastore')
+const errcode = require('err-code')
+
+const debug = require('debug')
+const log = debug('datastore-pubsub:publisher')
+log.error = debug('datastore-pubsub:publisher:error')
+
+// DatastorePubsub is responsible for providing an [interface-datastore]{@link https://github.com/ipfs/interface-datastore}
+// compliant api for pubsub.
+class DatastorePubsub {
+  /**
+   * Creates an instance of DatastorePubsub.
+   * @param {*} pubsub - pubsub implementation.
+   * @param {*} datastore - datastore instance.
+   * @param {*} peerId - peer-id instance.
+   * @param {Object} validator - validator functions.
+   * @param {function(record, peerId, callback)} validator.validate - function to validate a record.
+   * @param {function(received, current, callback)} validator.select - function to select the newest between two records.
+   * @memberof DatastorePubsub
+   */
+  constructor (pubsub, datastore, peerId, validator) {
+    this._pubsub = pubsub
+    this._datastore = datastore
+    this._peerId = peerId
+    this._validator = validator
+
+    // Bind _handleSubscription function, which is called by pubsub.
+    this._handleSubscription = this._handleSubscription.bind(this)
+  }
+
+  /**
+   * Publishes a value through pubsub.
+   * @param {Key} key identifier of the value to be published.
+   * @param {Buffer} val value to be propagated.
+   * @param {function(Error)} callback
+   * @returns {void}
+   */
+  put (key, val, callback) {
+    if (!(key instanceof Key)) {
+      const errMsg = `datastore key does not have a valid format`
+
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_INVALID_DATASTORE_KEY'))
+    }
+
+    if (!Buffer.isBuffer(val)) {
+      const errMsg = `received value is not a buffer`
+
+      log.error(errMsg)
+      return callback(errcode(new Error(errMsg), 'ERR_INVALID_VALUE_RECEIVED'))
+    }
+
+    const stringifiedTopic = key._buf.toString()
```

Comment:
Should just do `key.toString()` here so it's not accessing a "private" property.

Reply:
There are other instances of this that should be updated as well.

Reply:
I did not notice that `Key` had that function. Thanks 👍